diffsims/crystallography/_diffracting_vector.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2024 The diffsims developers
+#
+# This file is part of diffsims.
+#
+# diffsims is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# diffsims is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with diffsims.  If not, see <http://www.gnu.org/licenses/>.
+
+from diffsims.crystallography import ReciprocalLatticeVector
+import numpy as np
+from orix.vector.miller import _transform_space
+
+
+class DiffractingVector(ReciprocalLatticeVector):
+    r"""Reciprocal lattice vectors :math:`(hkl)` for use in electron
+    diffraction analysis and simulation.
+
+    All lengths are assumed to be given in Å or inverse Å.
+
+    This extends the :class:`ReciprocalLatticeVector` class.  Diffracting Vectors
+    focus on the subset of reciporical lattice vectors that are relevant for
+    electron diffraction based on the intersection of the Ewald sphere with the
+    reciprocal lattice.
+
+    This class is only used internally to store the DiffractionVectors generated from the
+    DiffractionSimulation class. It is not (currently) intended to be used directly by the user.
+
+    Parameters
+    ----------
+    phase : orix.crystal_map.Phase
+        A phase with a crystal lattice and symmetry.
+    xyz : numpy.ndarray, list, or tuple, optional
+        Cartesian coordinates of indices of reciprocal lattice vector(s)
+        ``hkl``. Default is ``None``. This, ``hkl``, or ``hkil`` is
+        required.
+    hkl : numpy.ndarray, list, or tuple, optional
+        Indices of reciprocal lattice vector(s). Default is ``None``.
+        This, ``xyz``, or ``hkil`` is required.
+    hkil : numpy.ndarray, list, or tuple, optional
+        Indices of reciprocal lattice vector(s), often preferred over
+        ``hkl`` in trigonal and hexagonal lattices. Default is ``None``.
+        This, ``xyz``, or ``hkl`` is required.
+    intensity : numpy.ndarray, list, or tuple, optional
+        Intensity of the diffraction vector(s). Default is ``None``.
+    rotation : orix.quaternion.Rotation, optional
```

Comment:
I don't understand why the vectors need to know which orientation took them to where they are now. Can you explain?

Reply:
I think I was trying a bit too hard to follow the `RLV`. The problem is that once you apply a rotation to a RLV, the hkl no longer makes sense. In essence, you rotate the entire Lattice object, which we could do, but if we want to be able to multiple by a `Rotation` object we have to change how we think about that.  (See #211)

Reply:
Thank you for explaining. Given this class is private, I'm OK this design, although we have taken a different approach in kikuchipy.

I think you're here at the heart of what we've discussed in this PR (although I didn't know at first!). In kikuchipy, again, we've avoided this problem entirely by composition. A simulation object stores the perfect, pristine, and un-rotated reciprocal lattice vectors (hkl), the crystal rotations, and the corresponding reflections *on the detector*, i.e. *not by themselves inside a rotated crystal*. Each reflection on the detector points back to one of the un-rotated reciprocal lattice vectors.